ext/pdo_pgsql/pgsql_driver.c

```diff
@@ -1224,6 +1242,34 @@ PHP_METHOD(PDO_PGSql_Ext, pgsqlGetPid)
 }
 /* }}} */
 
+/* {{{ proto bool PDO::pgsqlSetNoticeCallback(mixed callback)
+   Sets a callback to receive DB notices (after client_min_messages has been set) */
+PHP_METHOD(PDO_PGSql_Ext, pgsqlSetNoticeCallback)
+{
+	pdo_dbh_t *dbh;
+	pdo_pgsql_db_handle *H;
+	zend_fcall_info fci = empty_fcall_info;
+	zend_fcall_info_cache fcc = empty_fcall_info_cache;
+	
+	dbh = Z_PDO_DBH_P(getThis());
+	PDO_CONSTRUCT_CHECK;
```

Comment:
This check should only be done _after_ parameter parsing succeeded.

Reply:
OK, I combined declaration and initialization of the `pdo_dbh_t *` and `pdo_pgsql_db_handle *`, so ZPP does not even know we're in a PDO context. Is this OK, or should I at least have `Z_PDO_DBH_P(ZEND_THIS)` before ZPP (itself before PDO_CONSTRUCT_CHECK)?

Reply:
This is ok as it is now, the placement of dbh doesn't really matter.